test/emu_sv/test_sparse_operator.py

```diff
@@ -0,0 +1,166 @@
+import torch
+import pytest
+from functools import reduce
+
+from emu_sv import StateVector, SparseOperator
+from emu_sv.sparse_operator import sparse_kron, sparse_add
+
+dtype = torch.complex128
+X = torch.tensor([[0, 1], [1, 0]], dtype=dtype)
+Y = torch.tensor([[0, -1j], [1j, 0]], dtype=dtype)
+Z = torch.tensor([[1, 0], [0, -1]], dtype=dtype)
+Id = torch.tensor([[1, 0], [0, 1]], dtype=dtype)
+zero_state_torch = torch.tensor([[1, 0]], dtype=dtype)
+one_state_torch = torch.tensor([[0, 1]], dtype=dtype)
+
+
+@pytest.mark.parametrize(("zero", "one"), [("g", "r")])
+def test_from_operator_repr_and_rmul(zero: str, one: str) -> None:
+    # creation 2 qubit operator X_0Z_1
+    N = 2
+    operations = [
+        (
+            1.0,
+            [
+                ({zero + one: 1.0, one + zero: -1.0}, {0}),  # iY
+                ({zero + zero: 1.0, one + one: -1.0}, {1}),  # Z
+            ],
+        )
+    ]
+
+    operator = SparseOperator.from_operator_repr(
+        eigenstates=(one, zero),
+        n_qudits=N,
+        operations=operations,
+    )
+    expected = torch.kron(1j * Y, Z)
+    assert torch.allclose(operator.matrix.to_dense().cpu(), expected)
+
+    # multiplying an operator by a number
+    operator = 3 * operator  # testing __rmul__
+    expected = 3 * expected
+    assert torch.allclose(operator.matrix.to_dense().cpu(), expected)
+
+
+@pytest.mark.parametrize(("zero", "one"), [("g", "r")])
```

Comment:
Do we really need parametrisation ? its a for loop with a single element

the same for the next tests